diff --git a/mozglue/build/moz.build b/mozglue/build/moz.build
index 53758485ac3d..3b5ce063ebb4 100644
--- a/mozglue/build/moz.build
+++ b/mozglue/build/moz.build
@@ -9,6 +9,7 @@
 # If this is ever changed, update MOZ_SHARED_MOZGLUE in browser/installer/Makefile.in
 if CONFIG['JS_STANDALONE'] and not CONFIG['MOZ_MEMORY']:
     Library('mozglue')
+    FINAL_LIBRARY = 'js'
 elif CONFIG['OS_TARGET'] in ('WINNT', 'Darwin', 'Android'):
     SharedLibrary('mozglue')
 else:
