diff --git a/mozglue/build/moz.build b/mozglue/build/moz.build
index 58e2db6..1cfb504 100644
--- a/mozglue/build/moz.build
+++ b/mozglue/build/moz.build
@@ -4,12 +4,9 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-# Build mozglue as a shared lib on Windows, OSX and Android.
-# If this is ever changed, update MOZ_SHARED_MOZGLUE in browser/installer/Makefile.in
-if CONFIG['OS_TARGET'] in ('WINNT', 'Darwin', 'Android'):
-    SharedLibrary('mozglue')
-else:
-    Library('mozglue')
+# Build mozglue as a static lib into js for 0 A.D.
+Library('mozglue')
+FINAL_LIBRARY = 'js'
 
 if not CONFIG['MOZ_CRT']:
     SDK_LIBRARY = True
