--- a/build/moz.configure/toolchain.configure
+++ b/build/moz.configure/toolchain.configure
@@ -1619,8 +1619,9 @@
         # Initialize local variables with a 0xAA pattern in clang debug builds.
         # Linux32 fails some xpcshell tests with -ftrivial-auto-var-init
         linux32 = target.kernel == 'Linux' and target.cpu == 'x86'
+        mac = target.kernel == 'Darwin'
         if (c_compiler.type == 'clang' or c_compiler.type == 'clang-cl') and \
-            c_compiler.version >= '8' and debug and not linux32:
+            c_compiler.version >= '8' and debug and not linux32 and not mac:
             if c_compiler.type == 'clang-cl':
                 flags.append('-Xclang')
                 js_flags.append('-Xclang')
--- a/build/moz.configure/toolchain.configure
+++ b/build/moz.configure/toolchain.configure
@@ -141,7 +141,6 @@
     @imports(_from='biplist', _import='readPlist')
     def macos_sdk(sdk, host):
         sdk_min_version = Version('10.11')
-        sdk_max_version = Version('10.15.4')
 
         if sdk:
             sdk = sdk[0]
@@ -170,11 +169,6 @@
                 'You may need to point to it using --with-macos-sdk=<path> in your '
                 'mozconfig. Various SDK versions are available from '
                 'https://github.com/phracker/MacOSX-SDKs' % (version, sdk_min_version))
-        if version > sdk_max_version:
-            die('SDK version "%s" is unsupported. Please downgrade to version '
-                '%s. You may need to point to it using --with-macos-sdk=<path> in '
-                'your mozconfig. Various SDK versions are available from '
-                'https://github.com/phracker/MacOSX-SDKs' % (version, sdk_max_version))
         return sdk
 
     set_config('MACOS_SDK_DIR', macos_sdk)