diff --git a/js/public/RootingAPI.h b/js/public/RootingAPI.h
index 4e925cba04..06d2a95440 100644
--- a/js/public/RootingAPI.h
+++ b/js/public/RootingAPI.h
@@ -1019,6 +1019,7 @@ class PersistentRooted : public js::PersistentRootedBase<T>,
         MOZ_ASSERT(kind == js::THING_ROOT_OBJECT ||
                    kind == js::THING_ROOT_SCRIPT ||
                    kind == js::THING_ROOT_STRING ||
+                   kind == js::THING_ROOT_SYMBOL ||
                    kind == js::THING_ROOT_ID ||
                    kind == js::THING_ROOT_VALUE ||
                    kind == js::THING_ROOT_TRACEABLE);

