diff --git a/js/src/old-configure b/js/src/old-configure
index 7d621b52ce..85e2934fa9 100644
--- a/js/src/old-configure
+++ b/js/src/old-configure
@@ -7013,11 +7013,11 @@
 
 
 
-if test -n "$JS_STANDALONE"; then
 MOZ_APP_NAME="mozjs"
-JS_LIBRARY_NAME="mozjs-$MOZILLA_SYMBOLVERSION"
-else
-JS_LIBRARY_NAME="mozjs"
+if test -n "$MOZ_DEBUG"; then
+JS_LIBRARY_NAME="mozjs$MOZILLA_SYMBOLVERSION-ps-debug"
+else
+JS_LIBRARY_NAME="mozjs$MOZILLA_SYMBOLVERSION-ps-release"
 fi
 JS_CONFIG_LIBS="$NSPR_LIBS $LIBS"
 if test -n "$GNU_CC"; then
