Index: libraries/nvtt/src/CMakeLists.txt
===================================================================
--- libraries/nvtt/src/CMakeLists.txt	(revision 8295)
+++ libraries/nvtt/src/CMakeLists.txt	(working copy)
@@ -22,6 +22,10 @@
 	SET(NVIMAGE_SHARED TRUE)
 ENDIF(NVTT_SHARED)
 
+SET(CMAKE_SKIP_BUILD_RPATH TRUE)
+SET(CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
+SET(CMAKE_INSTALL_RPATH "$ORIGIN")
+SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 
 ADD_SUBDIRECTORY(src)
 
