Index: src/src/nvcore/nvcore.h
===================================================================
--- src/src/nvcore/nvcore.h	(revision 13633)
+++ src/src/nvcore/nvcore.h	(revision 13634)
@@ -67,6 +67,7 @@
 // NV_CPU_X86
 // NV_CPU_X86_64
 // NV_CPU_PPC
+// NV_CPU_ARM
 
 #define NV_CPU_STRING 	POSH_CPU_STRING
 
@@ -76,6 +77,8 @@
 #	define NV_CPU_X86 1
 #elif defined POSH_CPU_PPC
 #	define NV_CPU_PPC 1
+#elif defined POSH_CPU_STRONGARM
+#	define NV_CPU_ARM 1
 #else
 #	error "Unsupported CPU"
 #endif
