Index: src/CMakeLists.txt
===================================================================
--- src/CMakeLists.txt	(revision 13168)
+++ src/CMakeLists.txt	(working copy)
@@ -9,12 +9,15 @@
 # initial variables
 SET(GLUT TRUE CACHE BOOL "")
 SET(GLEW TRUE CACHE BOOL "")
-SET(CG TRUE CACHE BOOL "")
-SET(CUDA TRUE CACHE BOOL "")
-SET(OPENEXR TRUE CACHE BOOL "")
-
-# OpenGL
-INCLUDE(FindOpenGL)
+SET(CG TRUE CACHE BOOL "")
+SET(CUDA TRUE CACHE BOOL "")
+SET(OPENEXR TRUE CACHE BOOL "")
+SET(JPEG TRUE CACHE BOOL "")
+SET(PNG TRUE CACHE BOOL "")
+SET(TIFF TRUE CACHE BOOL "")
+
+# OpenGL
+INCLUDE(FindOpenGL)
 IF(OPENGL_FOUND)
 	MESSAGE(STATUS "Looking for OpenGL - found")
 ELSE(OPENGL_FOUND)
@@ -78,37 +81,43 @@
 	MESSAGE(STATUS "Looking for Maya - found")
 ELSE(MAYA_FOUND)
 	MESSAGE(STATUS "Looking for Maya - not found")
-ENDIF(MAYA_FOUND)
-
-# JPEG
-INCLUDE(FindJPEG)
-IF(JPEG_FOUND)
-	SET(HAVE_JPEG ${JPEG_FOUND} CACHE BOOL "Set to TRUE if JPEG is found, FALSE otherwise")
-	MESSAGE(STATUS "Looking for JPEG - found")
-ELSE(JPEG_FOUND)
-	MESSAGE(STATUS "Looking for JPEG - not found")
-ENDIF(JPEG_FOUND)
-
-# PNG
-INCLUDE(FindPNG)
-IF(PNG_FOUND)
-	SET(HAVE_PNG ${PNG_FOUND} CACHE BOOL "Set to TRUE if PNG is found, FALSE otherwise")
-	MESSAGE(STATUS "Looking for PNG - found")
-ELSE(PNG_FOUND)
-	MESSAGE(STATUS "Looking for PNG - not found")
-ENDIF(PNG_FOUND)
-
-# TIFF
-INCLUDE(FindTIFF)
-IF(TIFF_FOUND)
-	SET(HAVE_TIFF ${TIFF_FOUND} CACHE BOOL "Set to TRUE if TIFF is found, FALSE otherwise")
-	MESSAGE(STATUS "Looking for TIFF - found")
-ELSE(TIFF_FOUND)
-	MESSAGE(STATUS "Looking for TIFF - not found")
-ENDIF(TIFF_FOUND)
-
-# OpenEXR
-IF(OPENEXR)
+ENDIF(MAYA_FOUND)
+
+# JPEG
+IF(JPEG)
+	INCLUDE(FindJPEG)
+	IF(JPEG_FOUND)
+		SET(HAVE_JPEG ${JPEG_FOUND} CACHE BOOL "Set to TRUE if JPEG is found, FALSE otherwise")
+		MESSAGE(STATUS "Looking for JPEG - found")
+	ELSE(JPEG_FOUND)
+		MESSAGE(STATUS "Looking for JPEG - not found")
+	ENDIF(JPEG_FOUND)
+ENDIF(JPEG)
+
+# PNG
+IF(PNG)
+	INCLUDE(FindPNG)
+	IF(PNG_FOUND)
+		SET(HAVE_PNG ${PNG_FOUND} CACHE BOOL "Set to TRUE if PNG is found, FALSE otherwise")
+		MESSAGE(STATUS "Looking for PNG - found")
+	ELSE(PNG_FOUND)
+		MESSAGE(STATUS "Looking for PNG - not found")
+	ENDIF(PNG_FOUND)
+ENDIF(PNG)
+
+# TIFF
+IF(TIFF)
+	INCLUDE(FindTIFF)
+	IF(TIFF_FOUND)
+		SET(HAVE_TIFF ${TIFF_FOUND} CACHE BOOL "Set to TRUE if TIFF is found, FALSE otherwise")
+		MESSAGE(STATUS "Looking for TIFF - found")
+	ELSE(TIFF_FOUND)
+		MESSAGE(STATUS "Looking for TIFF - not found")
+	ENDIF(TIFF_FOUND)
+ENDIF(TIFF)
+
+# OpenEXR
+IF(OPENEXR)
 	INCLUDE(${NV_CMAKE_DIR}/FindOpenEXR.cmake)
 	IF(OPENEXR_FOUND)
 		SET(HAVE_OPENEXR ${OPENEXR_FOUND} CACHE BOOL "Set to TRUE if OpenEXR is found, FALSE otherwise")
