---
 src/nvtt/CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/nvtt/CMakeLists.txt b/src/nvtt/CMakeLists.txt
index df77c86..e543807 100644
--- a/src/nvtt/CMakeLists.txt
+++ b/src/nvtt/CMakeLists.txt
@@ -41,7 +41,11 @@ INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR})
 
 ADD_DEFINITIONS(-DNVTT_EXPORTS)
 
-IF(NVTT_SHARED)	
+IF(NVTT_SHARED)
+    IF(CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
+        SET(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-z,origin")
+    ENDIF(CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
+
     ADD_LIBRARY(nvtt SHARED ${NVTT_SRCS})
 ELSE(NVTT_SHARED)
     ADD_LIBRARY(nvtt ${NVTT_SRCS})
--
