diff --git "a/src/nvmath/Half.cpp" "b/src/nvmath/Half.cpp"
index 9021771..bf37d3e 100644
--- "a/src/nvmath/Half.cpp"
+++ "b/src/nvmath/Half.cpp"
@@ -492,7 +492,9 @@ nv::half_to_float( uint16 h )
 #if !NV_OS_IOS && (defined(__i386__) || defined(__x86_64__))

 #if NV_CC_GNUC
-#if defined(__i386__) || defined(__x86_64__)
+#if defined(__i386__)
+#include <x86intrin.h>
+#elif defined(__x86_64__)
 #include <xmmintrin.h>
 #endif
 #endif
