Description: Allow building 0ad as root/fakeroot
Forwarded: not-needed
Author: Rico Tzschichholz <ricotz@ubuntu.com>
Last-Update: 2014-10-13

--- a/build/workspaces/update-workspaces.sh
+++ b/build/workspaces/update-workspaces.sh
@@ -1,10 +1,5 @@
 #!/bin/sh
 
-if [ "$(id -u)" = "0" ]; then
-   echo "Running as root will mess up file permissions. Aborting ..." 1>&2
-   exit 1
-fi
-
 die()
 {
   echo ERROR: $*
