From: Nicolas Auvray <na.itms76@gmail.com>
Date: Thu, 13 Aug 2020 18:27:44 +0200
Subject: Fix #949699 on 0.0.23b
Description: Workaround for L3 cache detection of Ryzen 3000
Bug: https://trac.wildfiregames.com/ticket/4360
Bug-Debian: https://bugs.debian.org/949699

--- a/source/lib/sysdep/arch/x86_x64/cache.cpp
+++ b/source/lib/sysdep/arch/x86_x64/cache.cpp
@@ -89,7 +89,8 @@
 static const size_t associativityTable[16] =
 {
 	0, 1, 2, 0, 4, 0, 8, 0,
-	16, 0, 32, 48, 64, 96, 128, x86_x64::Cache::fullyAssociative
+	// TODO: The second '16' does not obey to the specifications and is only a workaround. For a correct implementation please look here: https://community.amd.com/thread/244207
+	16, 16, 32, 48, 64, 96, 128, x86_x64::Cache::fullyAssociative
 };
 
 static x86_x64::Cache L2Cache(u32 reg, x86_x64::Cache::Type type)
