--- js-1.8.5/js/src/configure.in
+++ js-1.8.5/js/src/configure.in
@@ -3041,6 +3041,9 @@
 *-freebsd*|*-kfreebsd*)
     AC_DEFINE(AVMPLUS_UNIX)
     ;;
+*-openbsd*)
+    AC_DEFINE(AVMPLUS_UNIX)
+    ;;
 *-gnu*)
     AC_DEFINE(AVMPLUS_UNIX)
     ;;
--- js-1.8.5/js/src/configure
+++ js-1.8.5/js/src/configure
@@ -8093,6 +8093,12 @@
 EOF
 
     ;;
+*-openbsd*)
+    cat >> confdefs.h <<\EOF
+#define AVMPLUS_UNIX 1
+EOF
+
+    ;;
 *-gnu*)
     cat >> confdefs.h <<\EOF
 #define AVMPLUS_UNIX 1
