Index: extern/poshlib/posh.h
===================================================================
--- extern/poshlib/posh.h	(revision 1024)
+++ extern/poshlib/posh.h	(revision 1025)
@@ -293,6 +293,11 @@
 #  define POSH_OS_STRING "Linux"
 #endif
 
+#if defined __FreeBSD__
+#  define POSH_OS_FREEBSD 1 
+#  define POSH_OS_STRING "FreeBSD"
+#endif
+
 #if defined __CYGWIN32__
 #  define POSH_OS_CYGWIN32 1
 #  define POSH_OS_STRING "Cygwin"
