diff --git a/include/fmt/os.h b/include/fmt/os.h
index 88151006..11c3b2cc 100644
--- a/include/fmt/os.h
+++ b/include/fmt/os.h
@@ -375,7 +375,7 @@ struct ostream_params {
 };
 }  // namespace detail
 
-static constexpr detail::buffer_size buffer_size;
+static constexpr detail::buffer_size buffer_size{};
 
 // A fast output stream which is not thread-safe.
 class ostream final : private detail::buffer<char> {
