Index: atomicrefcount.cpp
===================================================================
--- src/atomicrefcount.cpp	(revision 4528)
+++ src/atomicrefcount.cpp	(revision 4529)
@@ -76,7 +76,7 @@
 #if defined( _WIN32 ) && !defined( __SYMBIAN32__ )
       ::InterlockedExchange( (volatile LONG*)&m_count, (volatile LONG)0 );
 #elif defined( __APPLE__ )
-      OSAtomicAnd32Barrier( (int32_t)0, (volatile int32_t*)&m_count );
+      OSAtomicAnd32Barrier( (uint32_t)0, (volatile uint32_t*)&m_count );
 #elif defined( HAVE_GCC_ATOMIC_BUILTINS )
       // Use the gcc intrinsic for atomic decrement if supported.
       __sync_fetch_and_and( &m_count, 0 );
